resolwe/observers/views.py

```diff
@@ -0,0 +1,55 @@
+"""Background task viewset."""
+
+from rest_framework import mixins, serializers, viewsets
+
+from resolwe.observers.mixins import ObservableMixin
+from resolwe.observers.models import BackgroundTask
+
+STATUSES = [status[0] for status in BackgroundTask.STATUS_CHOICES]
+
+
+class BackgroundTaskOptions(serializers.Serializer):
+    """Used to deserialize options set no BackgrondTaskViewSet."""
+
+    with_superuser = serializers.BooleanField(default=False, required=False)
+    statuses = serializers.MultipleChoiceField(
+        choices=STATUSES,
+        required=False,
+        default=[BackgroundTask.STATUS_PREPARING, BackgroundTask.STATUS_PROCESSING],
+    )
+
+
+class BackgroundTaskSerializer(serializers.ModelSerializer):
+    class Meta:
+        """Set the model."""
+
+        model = BackgroundTask
+        fields = ("pk", "started", "finished", "status", "task", "output")
+
+
+class BackgroundTaksViewSet(
+    ObservableMixin, mixins.ListModelMixin, viewsets.GenericViewSet
+):
+    """Observable task viewset.
+
+    It exposes standard observable endpoints so user can subscribe to it plus it lists
+    all the background tasks for a current user.
+    """
+
+    serializer_class = BackgroundTaskSerializer
+
+    def get_queryset(self):
```

Comment:
You just need to filter by the user once the filter is added.

Reply:
I have used the resolwe permission backend to do the filtering.